src/server.c

```diff
@@ -4697,25 +4721,98 @@ sds genRedisInfoStringCommandStats(sds info, dict *commands) {
     return info;
 }
 
+void addSectionsToDict(dict *section_dict, char **sections, int len) {
+    for (int i = 0; i < len; i++){
+        sds section = sdsnew(sections[i]);
+        dictAdd(section_dict, section, NULL);
+    }
+}
+
+/* Create a dictionary according to the user input or call function 
+   The input parameter **argv indicate the 2D array which could includes multiply string,
+   abd argc indicates the 2D array length
+
+   Another two parameter *out_all and *out)everything will be set value in this function based 
+   on the argv parameter.
+
+   The return value will be the dictionary for genRedisInfoString function
+ */
+dict *genInfoSectionDict(robj **argv, int argc, int *out_all, int *out_everything) {
+    char *defSections[] = {"server", "clients", "memory", "persistence", "stats", "replication", "cpu", "modules", "errorstats", "cluster", "keyspace", "latencystats"};
+    char *defSectionsSentinel[] = {"server", "clients", "cpu", "stats", "sentinel"};
+
+
+    dict *section_dict = dictCreate(&stringSetDictType);
+    if (server.sentinel_mode) {
+        if (argc == 1) {
+            addSectionsToDict(section_dict, defSectionsSentinel, sizeof(defSectionsSentinel)/sizeof(*defSectionsSentinel));
+        } else {
+            for (int i = 1; i < argc; i++) {
+                if (!strcasecmp(argv[i]->ptr,"default")) 
+                    addSectionsToDict(section_dict, defSectionsSentinel, sizeof(defSectionsSentinel)/sizeof(*defSectionsSentinel));
+                else if (!strcasecmp(argv[i]->ptr,"all")) 
+                    (*out_all) = 1;
+                else {
+                    sds section = sdsnew(argv[i]->ptr);
+                    sdstolower(section);
+                    dictAdd(section_dict,section,NULL);
+                }
+            }
+        }
+    } else {
+        if (argc == 1) {
+            if (!strcasecmp(argv[0]->ptr,"info") || !strcasecmp(argv[0]->ptr,"default")) 
+                addSectionsToDict(section_dict, defSections, sizeof(defSections)/sizeof(*defSections));
+            else if (!strcasecmp(argv[0]->ptr,"all")) 
+                    (*out_all) = 1;
+                else if (!strcasecmp(argv[0]->ptr,"everything")) 
+                    (*out_everything) = 1;
+            else {  
+                sds section = sdsnew(argv[0]->ptr);
+                sdstolower(section);
+                dictAdd(section_dict,section,NULL);
+                if (!strcasecmp(argv[0]->ptr,"modules")) {
+                    sds section = sdsnew("input-modules");
```

Comment:
I want to explain why I add another fake input "input-modules" here.

Below is the output summary when user input 5 different cases:

![image](https://user-images.githubusercontent.com/51993843/148443250-ebdb2103-7f57-4a68-91b2-719a7dab73c0.png)